src/NServiceBus.NHibernate/SharedSession/SharedSessionConfig.cs

```diff
@@ -0,0 +1,21 @@
+namespace NServiceBus.Persistence.NHibernate
+{
+    using Configuration.AdvanceExtensibility;
+    using NServiceBus.Features;
+
+    /// <summary>
+    /// Deduplication configuration extensions.
+    /// </summary>
+    public static class SharedSessionConfig
+    {
+        /// <summary>
+        /// Disables sharing a connection between SQL Server transport and NHibernate persistence. Intended to be used in conjunction with Outbox only.
+        /// </summary>
+        /// <param name="persistenceConfiguration"></param>
+        public static PersistenceExtentions<NHibernatePersistence> DoNotShareTransportConnection(this PersistenceExtentions<NHibernatePersistence> persistenceConfiguration)
```

Comment:
Very confusing API IMO.
When would a user need to turn this on/off ?
You mention "Intended to be used in conjunction with Outbox only.", the api as it stands does not give any clue apart from the api doco, that is the intent.

Maybe we should consider making this setting part of the Outbox API configuration, eg:
```c#
config.UsePersistence<NHibernate, StorageType.Outbox>()
                                                                .DoNotShareTransportConnection();
```